src/CMakeLists.txt

```diff
@@ -738,100 +738,101 @@ if (${NUPIC_BUILD_PYEXT_MODULES})
   endfunction(PREPEND_BOILERPLATE_TO_PYTHON_PROXY_MODULE)
 
 
+  function(BUILD_EXTENSION MODULE_NAME)
```

Comment:
With the exception of the new code that creates the export map, the rest was scrubbed from existing boilerplate that used to be replicated for each extension below. Now, each one simply calls this shared function.